sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/DynamicDestinations.java

```diff
@@ -142,7 +142,11 @@ void setSideInputAccessorFromProcessContext(DoFn<?, ?>.ProcessContext context) {
     return null;
   }
 
-  /** Returns a {@link TableDestination} object for the destination. May not return null. */
+  /**
+   * Returns a {@link TableDestination} object for the destination. May not return null. Return
+   * value needs to be unique to each destination: may not return the same {@link TableDestination}
+   * for different destinations.
```

Comment:
Should we update Python SDK as well ?

cc: @pabloem

Reply:
Hm I do not think this is necessary in Python. In Python, we expect table names to be returned as the destination, so this case may not occur...

Reply:
TLDR: Pablo is right.

In Python SDK, a user function translates an element to a TableReference.
In Java SDK, a user DynamicDestinations instance translates an element to a DestinationT, and then to a TableDestination.

Java does a reshuffle on (DestinationT, element) pairs, while Python does it on (TableReference, element) pairs.

(Not sure why Java uses an intermediate DestinationT. Convenience? Better GBK performance? Lower resource use?)